uiowa/uiowa
# Feature: card option updates

Resolves: https://github.com/uiowa/uiowa/issues/2306

# How to test
*If you get an error like `Error: File to import not found or unreadable`, you need to run the following commands:*
- `rm -rf ./node_modules`
- `yarn cache clean`

Steps:
- `blt frontend`
- **Test update hook:** test on `v3` sites with vertical card rows (https://uiowa.edu/academics/areas-study, https://uiowa.edu/academics/leading-programs, https://uiowa.edu/academics/career-preparation) and horizontal card stacks (https://uiowa.edu/admissions, https://coronavirus.uiowa.edu/health). 
- **Existing card implementations:**  test on `v2` and `v3` article lists, people lists, featured content lists, and cards. 
- **New display options:** test all the new card display options using the new display options. Check to make sure that hiding and showing of the image size column works as described below in 'Programmatic updates' and that, if you do trick the form into allowing you to enter a forbidden combination, it is not saved.

# Updates 

## Programmatic updates

- Programmatically setting the card image style to be square instead of widescreen when the circle/round 1:1 option is selected
- Hiding the image size option when the stacked image option is selected AND either the square or widescreen image format is selected
- Form submission logic to reset image size to large if the user selects a different size when the option is shown, but then changes other options that would no longer allow it.

## Card options

**Image position**  - select one
- Stacked (default)
- Left - this and the next option replace the `horizontal` option.
- Right

**Image format** - select one
- Widescreen (16:9) (default)
- Square
- Circle

**Image size** - select one
- Small 
- Medium
- Large (default)

**Background color** - select one
- Default is still a white background
- Only `gray` currently is an option

**General Styles** - select any
- Border
